File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/splash/ISplashActivity.java
package openfoodfacts.github.scrachx.openfood.views.splash;

/**
 * Created by Lobster on 03.03.18.
 */
public interface ISplashActivity {
    interface Controller {
        void refreshData();
    }

    interface View {
        void showLoading();

        void hideLoading(boolean isError);

        void navigateToMainActivity();
    }
}

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/splash/ISplashActivity.kt
package openfoodfacts.github.scrachx.openfood.features.splash

import io.reactivex.disposables.Disposable

/**
 * Created by Lobster on 03.03.18.
 */
interface ISplashActivity {
    interface Controller : Disposable {
        fun refreshData()
    }

    interface View {
        fun showLoading()
        fun hideLoading(isError: Boolean)
        fun navigateToMainActivity()
    }
}